interp/notation_ops.ml

```diff
@@ -1090,7 +1090,7 @@ let rec match_ inner u alp metas sigma a1 a2 =
      otherwise how to ensure it corresponds to a well-typed eta-expansion;
      we make an exception for types which are metavariables: this is useful e.g.
      to print "{x:_ & P x}" knowing that notation "{x & P x}" is not defined. *)
-  | b1, NLambda (Name id as na,(NHole _ | NVar _ as t2),b2) when inner ->
+  | b1, NLambda (Name id as na,_,(NHole _ | NVar _ as t2),b2) when inner ->
```

Comment:
For the eta-expansion, one need to match exactly an `Explicit` for the expansion to make sense.

Typical situation: we don't want to identify `let y := fun {A} => g A in t` with `let y := g in t` as this would break the implicit argument of `y`.